search_params.go

```diff
@@ -0,0 +1,100 @@
+package stripe
+
+import (
+	"context"
+)
+
+//
+// Public types
+//
+
+// SearchContainer is a general interface for which all search result object structs
+// should comply. They achieve this by embedding a SearchMeta struct and
+// inheriting its implementation of this interface.
+type SearchContainer interface {
+	GetSearchMeta() *SearchMeta
+}
+
+// SearchMeta is the structure that contains the common properties of the search iterators
+type SearchMeta struct {
+	HasMore  bool    `json:"has_more"`
+	NextPage *string `json:"next_page"`
+	URL      string  `json:"url"`
+}
+
+// GetSearchMeta returns a SearchMeta struct (itself). It exists because any
+// structs that embed SearchMeta will inherit it, and thus implement the
+// SearchContainer interface.
+func (l *SearchMeta) GetSearchMeta() *SearchMeta {
+	return l
+}
+
+// SearchParams is the structure that contains the common properties
+// of any *SearchParams structure.
+type SearchParams struct {
+	// Context used for request. It may carry deadlines, cancelation signals,
+	// and other request-scoped values across API boundaries and between
+	// processes.
+	//
+	// Note that a cancelled or timed out context does not provide any
+	// guarantee whether the operation was or was not completed on Stripe's API
+	// servers. For certainty, you must either retry with the same idempotency
+	// key or query the state of the API.
+	Context context.Context `form:"-"`
+
+	Query        string    `form:"query"`
+	Expand       []*string `form:"expand"`
+	SearchWindow *string   `form:"search_window"`
+	SortOrder    *string   `form:"sort_order"`
+	Limit        *int64    `form:"limit"`
+	Page     	 *string   `form:"page"`
+
+	// Single specifies whether this is a single page iterator. By default,
+	// listing through an iterator will automatically grab additional pages as
+	// the query progresses. To change this behavior and just load a single
+	// page, set this to true.
+	Single bool `form:"-"` // Not an API parameter
+
+	// StripeAccount may contain the ID of a connected account. By including
+	// this field, the request is made as if it originated from the connected
+	// account instead of under the account of the owner of the configured
+	// Stripe key.
+	StripeAccount *string `form:"-"` // Passed as header
+}
+
+// GetSearchParams returns a SearchParams struct (itself). It exists because any
+// structs that embed SearchParams will inherit it, and thus implement the
+// SearchParamsContainer interface.
+func (p *SearchParams) GetSearchParams() *SearchParams {
+	return p
+}
+
+// GetParams returns SearchParams as a Params struct. It exists because any
+// structs that embed Params will inherit it, and thus implement the
+// ParamsContainer interface.
+func (p *SearchParams) GetParams() *Params {
+	return p.ToParams()
+}
+
+// SetStripeAccount sets a value for the Stripe-Account header.
+func (p *SearchParams) SetStripeAccount(val string) {
+	p.StripeAccount = &val
+}
```

Comment:
I don't think we actually need this for search infrastructure right?

Reply:
This is consistent with List params. A way to set stripe account on the search request.